From 9538f6b95a3be228edc078ce58258f0574142e0c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=C3=89rico=20Rolim?= <ericonr@disroot.org>
Date: Tue, 27 Oct 2020 11:17:41 -0300
Subject: [PATCH 248/267] Make elf.h header self contained.

The elf/elf.h header is shared, verbatim, by the elfutils project.
However, elfutils can be used on systems with libcs other than glibc,
making the presence of __BEGIN_DECLS, __END_DECLS and <features.h> in
the file something that downstream distros may have to add patches for.

Furthermore, this file doesn't declare anything with language linkage,
so `extern "C" {}` blocks aren't necessary; it also doesn't have any
conditional definitions based on feature test macros, making inclusion
of features.h unnecessary.
---
 elf/elf.h | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/elf/elf.h b/elf/elf.h
index ff9f1dadd5..6439c1a4d5 100644
--- a/elf/elf.h
+++ b/elf/elf.h
@@ -19,10 +19,6 @@
 #ifndef _ELF_H
 #define	_ELF_H 1
 
-#include <features.h>
-
-__BEGIN_DECLS
-
 /* Standard ELF types.  */
 
 #include <stdint.h>
@@ -4105,6 +4101,4 @@ enum
 #define R_ARC_TLS_LE_S9		0x4a
 #define R_ARC_TLS_LE_32		0x4b
 
-__END_DECLS
-
 #endif	/* elf.h */
-- 
2.29.0

